### PR TITLE
REF,ENH: refactor {BibTeX,Text}Output into Output class with subclasses

### LINKED ISSUE #68: Outputting to bibtex doesn't filter by used citations
Probably linked to #19 

Example output text

```
$duecredit summary --format text

DueCredit Report:
- Scientific tools library / numpy (v 1.10.4) [1]
- Scientific tools library / scipy (v 0.17) [2]
  - Single linkage hierarchical clustering / scipy.cluster.hierarchy:linkage (v 0.17) [3]
- Machine Learning library / sklearn (v 0.17) [4]

3 packages cited
0 modules cited
1 functions cited

References
----------

[1] Van Der Walt, S., Colbert, S.C. & Varoquaux, G., 2011. The NumPy array: a structure for efficient numerical computation. Computing in Science & Engineering, 13(2), pp.22–30.
[2] Jones, E. et al., 2001. SciPy: Open source scientific tools for Python.
[3] Sibson, R., 1973. SLINK: an optimally efficient algorithm for the single-link cluster method. The Computer Journal, 16(1), pp.30–34.
[4] Pedregosa, F. et al., 2011. Scikit-learn: Machine learning in Python. The Journal of Machine Learning Research, 12, pp.2825–2830.
```

Example output BibTeX

```
$duecredit summary --format bibtex
@article{sneath1962numerical,
        title={Numerical taxonomy},
        author={Sneath, Peter HA and Sokal, Robert R},
        journal={Nature},
        volume={193},
        number={4818},
        pages={855--860},
        year={1962},
        publisher={Nature Publishing Group}
    }
@article{batagelj1995comparing,
        title={Comparing resemblance measures},
        author={Batagelj, Vladimir and Bren, Matevz},
        journal={Journal of classification},
        volume={12},
        number={1},
        pages={73--90},
        year={1995},
        publisher={Springer}
    }
@Misc{JOP+01,
      author =    {Eric Jones and Travis Oliphant and Pearu Peterson and others},
      title =     {{SciPy}: Open source scientific tools for {Python}},
      year =      {2001--},
      url = "http://www.scipy.org/",
      note = {[Online; accessed 2015-07-13]}
    }

...
```

### PR DESCRIPTION
**Pull request by mvdoc**
This fixes #68, #48, and closes #83

### CODE REVIEW

**Comment by mvdoc** [comment]
fails on python 3.x. will debug later

**Comment by yarikoptic** [comment]
One of the failures seems to be just difference in indentation in wrapped title of the XXX0

**Comment by mvdoc** [comment]
ok all tests pass now

File: duecredit/collector.py [hunk 1]
```diff
@@ -161,8 +161,12 @@ def __contains__(self, entry):
             return entry.path.startswith(self.path + '.')
 
     @property
+    def entry_key(self):
+        return self.entry.get_key()
```
Comment (yarikoptic) [inline duecredit/collector.py]: Do we need it?
Why not to have key as a property in entry and then use `entry.key`

Comment (mvdoc) [inline duecredit/collector.py]: good point

**Pull request merged**